### ISSUE TITLE
Empty CognitoSignUpOptions() throws error [0.2.0-nullsafety.0]

### ISSUE THREAD

**Issue by Jordan-Nelson** [aws-amplify/amplify-flutter#626]
**Describe the bug**

In the latest null safe preview release (0.2.0-nullsafety.0), an empty CognitoSignUpOptions() throws an error.

**To Reproduce**
In app that uses email based auth (probably reproducible in a username based app, but have not tested) try to sign a user up with the following code

```dart
SignUpResult res = await Amplify.Auth.signUp(
  username: <EMAIL>,
  password: <PASSWORD>,
  options: CognitoSignUpOptions(), // leave CognitoSignUpOptions empty
);
```

**Expected behavior**
No error should be thrown

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[x] Android
[x] iOS

**Additional context**
CognitoSignUpOptions extends SignUpOptions. CognitoSignUpOptions takes a nullable userAttributes in the constructor. It passes it to SignUpOptions via a call to super(). userAttributes in NOT nullable in SignUpOptions . This means that if a null value is passed into CognitoSignUpOptions for userAttributes, it will then get passed to SignUpOptions and throw an error. Ideally this would be caught at compile time, but it is not. It seems like an issue w/ null safety and super().

**Comment by Jordan-Nelson** [comment]
After looking into #209 a little more, I think we probably just want to consider this a duplicate of #209. This can be resolved by changing options to `CognitoSignUpOptions(userAttributes: <String, String>{})`. It turns out this is already required on master, which I did not realize. A different error is thrown on master and the NS branch with an empty CognitoSignUpOptions(), but they are both equally unhelpful. 

If we fix 209, I think we will be resolving this as well.

**Comment by Jordan-Nelson** [comment]
Resolved in v0.2.1
